src/common/image.c

```diff
@@ -937,35 +939,19 @@ void dt_image_remove(const int32_t imgid)
   DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, imgid);
   sqlite3_step(stmt);
   sqlite3_finalize(stmt);
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), "DELETE FROM main.history_hash WHERE imgid = ?1",
+                              -1, &stmt, NULL);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, imgid);
+  sqlite3_step(stmt);
+  sqlite3_finalize(stmt);
   // also clear all thumbnails in mipmap_cache.
   dt_mipmap_cache_remove(darktable.mipmap_cache, imgid);
 }
 
 gboolean dt_image_altered(const uint32_t imgid)
 {
-  sqlite3_stmt *stmt;
-
-  const gboolean basecurve_auto_apply = dt_conf_get_bool("plugins/darkroom/basecurve/auto_apply");
-  const gboolean sharpen_auto_apply = dt_conf_get_bool("plugins/darkroom/sharpen/auto_apply");
-
-  char query[1024] = { 0 };
-
-  snprintf(query, sizeof(query),
-           "SELECT 1"
-           " FROM main.history, main.images"
-           " WHERE id=?1 AND imgid=id AND num<history_end AND enabled=1"
-           "       AND operation NOT IN ('flip', 'dither', 'highlights', 'rawprepare',"
-           "                             'colorin', 'colorout', 'gamma', 'demosaic', 'temperature'%s%s)",
-           basecurve_auto_apply ? ", 'basecurve'" : "",
-           sharpen_auto_apply ? ", 'sharpen'" : "");
-
-  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), query, -1, &stmt, NULL);
-  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, imgid);
-
-  const gboolean altered = (sqlite3_step(stmt) == SQLITE_ROW);
-  sqlite3_finalize(stmt);
-
-  return altered;
+  dt_hash_history_t status = dt_hash_history_get_status(imgid);
+  return status & DT_HH_CURRENT;
```

Comment:
with the suggested changes in enum this would just be:
`
return status == DT_HH_CURRENT;`

Reply:
So this is good given previous comment.